packages/grafana-schema/src/veneer/mudball.types.ts

```diff
@@ -0,0 +1,11 @@
+import * as raw from '../schema/mudball.gen';
+
+import { MatcherConfig } from './dashboard.types';
+
+export interface MapLayerOptions<TConfig = any> extends raw.MapLayerOptions {
+  // Custom options depending on the type
+  config?: TConfig;
+  filterData?: MatcherConfig;
+}
+
+export * from '../schema/mudball.gen';
```

Comment:
@sdboyer this seems to work, all the types defined above will overwrite the ones from mudball and the modified types will be exported. Haven't found any issues with this approach so far and it seems like nothing else is needed to make veneers work for mudball.